File: submissions/bits-please/challenge9.java
static void f(String s)
{
	int life=3;
	int big = 0;
	int star = 0;
	int coins=0;
	int end=1;
	String[] stories=s.replaceAll(" +", " ").split(" ");
	for(String story:stories)
	{
		if("goombakoopapiranha".contains(story)){
			if(star==0){
				if(big>0)
					big=0;
				else
					life--;
			}
		}
		if(star>0){
			star--;
		}
		if("Mushroom".equals(story)){
			big=1;
		}
		if("Star".equals(story)){
			star+=2;
		}
		if(story.matches("^-?\\d+$")){
			coins+=Integer.valueOf(story);
			life+=(int)(coins/100);
			coins%=100;
		}
		if("1Up".equals(story)){
			life++;
		}
		if(life>0){
			if("PrincessBowser".contains(story)){
				System.out.print("WIN");
				end--;
				break;
			}
		}else{
			System.out.print("GAME OVER");
			end--;
			break;
		}
	}
	if(end>0)System.out.print("???");
}

File: submissions/pfudor/challenge9.rb
def f(i)
s=[0,0,3,0]
t='???'
i.split.each do |h|
case h[0]
when 'p','g','k'
if s[0]<1
(s[1]==0)?s[2]-=1:s[1]=0
end
when 'B','P'
t='WIN'
break
end
if h[0]=='M'
s[1]=1
elsif h[0]=='S'
s[0]=3
elsif h=='1Up'
s[2]+=1
elsif !!h[/\d+/]
s[3]+=Integer(h)
if s[3]>=100
s[2]+=s[3]/100
s[3]=s[3]%100
end
end
if s[2]<=0
t='GAME OVER'
break
end
s[0]-=1
end
puts t
end